src/pages/EntityDetails/EntityDetails.js

```diff
@@ -29,8 +34,37 @@ import { fetchModelStatus } from "juju/actions";
 
 import "./_entity-details.scss";
 
-const EntityDetails = ({ activeView, setActiveView, type, children }) => {
-  const { modelName } = useParams();
+function generatePanelContent(activePanel, entity, panelRowClick) {
+  switch (activePanel) {
+    case "remoteApps":
+      return <RemoteAppsPanel entity={entity} panelRowClick={panelRowClick} />;
+    case "offers":
+      return <OffersPanel entity={entity} panelRowClick={panelRowClick} />;
+  }
+}
+
+const EntityDetails = ({ type, children }) => {
+  const modelStatusData = useModelStatus();
+  const { userName, modelName } = useParams();
+  const history = useHistory();
+
+  // Check if model exists in modelStatusData, else 404
+  if (modelName !== modelStatusData?.model?.name) {
```

Comment:
This makes it impossible to access a model before the application has had a chance to actually load the model data. To reproduce view a model then an app then refresh, it'll 404.